src/Reactor.cc

```diff
@@ -1036,6 +1049,40 @@ void RestoreMachineCommand::tabCompletion(vector<string>& tokens) const
 }
 
 
+// class SetupCommand
+
+SetupCommand::SetupCommand(CommandController& commandController_,
+                               Reactor& reactor_)
+	: Command(commandController_, "setup")
```

Comment:
I think we need a more descriptive command name than just `setup`.

Also do we want only the command to replace the current machine with a new setup? Or also the more low level commands to load a setup in an empty machine-tab?

Or maybe at a technical level we can just treat setups and savestates the same. Then we do already have all low level functionality. And then only add some higher level convenience functions on top?  (Maybe implemented in c++, maybe in Tcl).

Reply:
I had it more descriptive, but I thought it would be better to match the existing `machine` command. So, I took the same kind of terminology and the same kind of behaviour: it replaces the current machine with the specified setup.
For low level commands: we already have these, they are the same as for save state loading...

In other words: I think we should treat machines and setups the same, not setups and savestates.

Reply:
@m9710797 if you still want this to be changed, please let me know; if not, please close the thread.

Reply:
I do think `setup` is not a very descriptive command name.

I do understand you picked it to be consistent with the command `machine`. Though I also think `machine` is not a very descriptive command name. But for backwards compatibility we're stuck with that name (and it's not bad enough to warrant an extra command with a better name).

That all said, I don't feel strongly about this. If you think `setup` is good enough, then fine.

Reply:
We could change it later for both if we want and perhaps keep an old alias for machine. But I guess most people won't use the console command anyway.